kzg_prover/bin/gen_verifiers.rs

```diff
@@ -0,0 +1,149 @@
+        strategy::SingleStrategy,
+    },
+    transcript::TranscriptWriterBuffer,
+};
+use halo2_solidity_verifier::{
+    compile_solidity, encode_calldata, BatchOpenScheme::Bdfg21, Evm, Keccak256Transcript,
+    SolidityGenerator,
+};
+use num_bigint::BigUint;
+use prelude::*;
+use rand::rngs::OsRng;
+use summa_solvency::{
+    circuits::{
+        univariate_grand_sum::UnivariateGrandSum,
+        utils::{generate_setup_artifacts, open_grand_sums, verify_grand_sum_openings},
+    },
+    cryptocurrency::Cryptocurrency,
+    entry::Entry,
+    utils::{big_uint_to_fp, parse_csv_to_entries},
+};
+
+const K: u32 = 17;
+const N_CURRENCIES: usize = 2;
+const N_USERS: usize = 16;
+
+fn main() {
+    // In order to generate the verifier we create the circuit using the init_empty() method, which means that the circuit is not initialized with any data.
+    let circuit = UnivariateGrandSum::<N_USERS, N_CURRENCIES>::init_empty();
+
+    let (params, pk, _) =
+        generate_setup_artifacts(K, Some("../backend/ptau/hermez-raw-17"), &circuit).unwrap();
+
+    // Only now we can instantiate the circuit with the actual inputs
+    let mut entries: Vec<Entry<N_CURRENCIES>> = vec![Entry::init_empty(); N_USERS];
+    let mut cryptos = vec![Cryptocurrency::init_empty(); N_CURRENCIES];
+
+    parse_csv_to_entries::<&str, N_CURRENCIES>("../csv/entry_16.csv", &mut entries, &mut cryptos)
+        .unwrap();
+
+    let univariate_grand_sum_circuit =
+        UnivariateGrandSum::<N_USERS, N_CURRENCIES>::init(entries.to_vec());
+
+    // 1. Generate Snark Verifier Contract and Verification Key
+    //
+    // the instance value is not used in proving, but it is necessary to SolidityGenerator and it should at least 1.
+    let num_instance = 1_usize;
+    let generator: SolidityGenerator<'_> =
+        SolidityGenerator::new(&params, pk.get_vk(), Bdfg21, num_instance);
+    let (verifier_solidity, vk_verifier) = generator.render_separately().unwrap();
+
+    let verifier_solidity_fixed = verifier_solidity
+        .replace("Halo2Verifier", "Verifier")
+        .replace(") public returns (bool)", ") public view returns (bool)");
+
+    let verifier_code = compile_solidity(&verifier_solidity_fixed);
+    let vk_code = compile_solidity(&vk_verifier);
+
+    // 2. Generate Snark Proof for range check
+    //
+    // the instance values has to be at least more than one due to verifier contract that generated from SolidityGenerator.
+    let instances: Vec<Fr> = vec![Fp::one(); 1];
+    let mut transcript = Keccak256Transcript::new(Vec::new());
+
+    let result = create_proof::<_, ProverSHPLONK<_>, _, _, _, _>(
+        &params,
+        &pk,
+        &[univariate_grand_sum_circuit],
+        &[&[&instances]],
+        &mut OsRng,
+        &mut transcript,
+    );
+    assert!(result.is_ok());
+
+    let result_unwrapped = result.unwrap();
+    result_unwrapped.0.expect("prover should not fail");
+    let _advice_polys = result_unwrapped.1.clone();
```

Comment:
I guess these are the leftovers from the user inclusion verifier, could you please delete them?

Reply:
It removed!